File: src/Microsoft.Management.Configuration/DiagnosticInformation.cpp
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
#include "pch.h"
#include "DiagnosticInformation.h"
#include "DiagnosticInformation.g.cpp"

namespace winrt::Microsoft::Management::Configuration::implementation
{
    void DiagnosticInformation::Initialize(DiagnosticLevel level, std::wstring_view message)
    {
        m_level = level;
        m_message = message;
    }

    DiagnosticLevel DiagnosticInformation::Level()
    {
        return m_level;
    }

    void DiagnosticInformation::Level(DiagnosticLevel value)
    {
        m_level = value;
    }

    hstring DiagnosticInformation::Message()
    {
        return m_message;
    }

    void DiagnosticInformation::Message(const hstring& value)
    {
        m_message = value;
    }
}

File: src/Microsoft.Management.Configuration.Processor/Helpers/DiagnosticInformation.cs
// -----------------------------------------------------------------------------
// <copyright file="DiagnosticInformation.cs" company="Microsoft Corporation">
//     Copyright (c) Microsoft Corporation. Licensed under the MIT License.
// </copyright>
// -----------------------------------------------------------------------------

namespace Microsoft.Management.Configuration.Processor.Helpers
{
    using System;
    using Microsoft.Management.Configuration;

    /// <summary>
    /// Implements IDiagnosticInformation.
    /// </summary>
    internal sealed partial class DiagnosticInformation : IDiagnosticInformation
    {
        /// <inheritdoc/>
        public DiagnosticLevel Level { get; internal set; }

        /// <inheritdoc/>
        public string? Message { get; internal set; }
    }
}
